docs/source/export.rst

```diff
@@ -1,10 +1,445 @@
+.. _torch.export:
+
 torch.export
 =====================
 
-.. TODO: Add torch.export() tutorial here.
-
 .. warning::
-    This feature is a prototype and may have compatibility breaking changes in the future.
+    This feature is a prototype under active development and there WILL BE breaking changes in the future.
+
+
+Overview
+--------
+
+``torch.export`` is a one-shot process for a sound, full-program capture of a
+PyTorch model into a standardized representation ahead-of-time (AOT). It is
+designed to streamline and standardize the way PyTorch interacts with various
+systems and platforms.
+
+``torch.export`` maintains a clean intermediate representation (IR) with the
+following invariants. More specifications about the IR can be found here (coming
+soon!).
+
+* **Soundness**: It is guaranteed to be a sound representation of the original
+  program, and maintains the same calling conventions of the original program.
+
+* **Normalized**: There are no Python semantics within the graph. Submodules
+  from the original programs are inlined to form one fully flattened
+  computational graph.
+
+* **Defined Operator Set**: The graph produced contains only a small defined
+  :ref:`Core ATen IR <torch.compiler_ir>` opset and registered custom
+  operators.
+
+* **Graph properties**: The graph is purely functional, meaning it does not
+  contain operations with side effects such as mutations or aliasing. It does
+  not mutate any intermediate values, parameters, or buffers.
+
+* **Metadata**: The graph contains metadata captured during tracing, such as a
+  stacktrace from user's code.
+
+Under the hood, ``torch.export`` leverages the following latest technologies:
+
+* **TorchDynamo (torch._dynamo)** is an internal API that uses a CPython feature
+  called the Frame Evaluation API to safely trace PyTorch graphs. This
+  provides a massively improved graph capturing experience, with much fewer
+  rewrites needed in order to fully trace the PyTorch code.
+
+* **AOT Autograd** provides a functionalized PyTorch graph and ensures the graph
+  is decomposed/lowered to the small defined Core ATen operator set.
+
+* **Torch FX (torch.fx)** is the underlying representation of the graph,
+  allowing flexible Python-based transformations.
+
+
+Existing frameworks
+^^^^^^^^^^^^^^^^^^^
+
+:func:`torch.compile` also utilizes the same PT2 stack as ``torch.export``, but
+is slightly different in that it is a JIT compiler. When :func:`torch.compile`
+runs into an untraceable part of a model, it will "graph break" and fall back to
+running the program in the eager Python runtime. In comparison, ``torch.export``
+aims to get a full graph representation of a PyTorch model, so it will error out
+when something untraceable is reached. Since ``torch.export`` produces a full
+graph disjoint from any Python features or runtime, this graph can then be
+saved, loaded, and run in different environments and languages.
```

Comment:
Can we add a bullet point list in this paragraph highlighting the differences:

1) JIT vs AoT: torch.compile is not intended to be used to produce compiled artifact outside of deployment
2) partial vs full graph
3) Usability tradeoff (requires code rewriting vs not)